packages/playback-core/test/player.test.js

```diff
@@ -0,0 +1,169 @@
+import { assert, aTimeout } from '@open-wc/testing';
+import { initialize, teardown, MediaError, getError } from '../src/index.ts';
+
+describe('playback core', () => {
+  it('initialize w/ default preferPlayback', async function () {
+    const video = document.createElement('video');
+    const core = initialize(
+      {
+        src: 'https://stream.mux.com/23s11nz72DsoN657h4314PjKKjsF2JG33eBQQt6B95I.m3u8',
+      },
+      video
+    );
+    assert.equal(typeof core, 'object');
+    assert.equal(typeof core.engine, 'object');
+  });
+
+  it('teardown', async function () {
+    const video = document.createElement('video');
+    const core = initialize(
+      {
+        src: 'https://stream.mux.com/23s11nz72DsoN657h4314PjKKjsF2JG33eBQQt6B95I.m3u8',
+      },
+      video
+    );
+    teardown(video, core);
+
+    assert(!video.hasAttribute('src'), 'src is removed on teardown');
+  });
+
+  it('handle error', async function () {
+    const video = document.createElement('video');
+    initialize(
+      {
+        src: 'https://mux.com/', // not working src url
+      },
+      video
+    );
+    // error event must be attached after initialize(), otherwise the
+    // native error will not be stopped propagating in playback core.
+    await new Promise((resolve) => video.addEventListener('error', resolve));
+
+    assert.equal(video.error.code, MediaError.MEDIA_ERR_SRC_NOT_SUPPORTED);
+    assert.equal(video.error.message, '');
+
+    assert.equal(getError(video).code, MediaError.MEDIA_ERR_SRC_NOT_SUPPORTED);
+    assert.equal(
+      getError(video).message,
+      'An unsupported error occurred. The server or network failed, or your browser does not support this format.'
+    );
+  });
+
+  it('setAutoplay("any")', async function () {
+    this.timeout(10000);
+
+    const video = document.createElement('video');
+    const core = initialize(
+      {
+        src: 'https://stream.mux.com/23s11nz72DsoN657h4314PjKKjsF2JG33eBQQt6B95I.m3u8',
+        autoplay: false,
+      },
+      video
+    );
+
+    core.setAutoplay('any');
+    await new Promise((resolve) => video.addEventListener('playing', resolve));
```

Comment:
nitpick(non-blocking): I believe the assertion below is redundant, since `playing` implies `!paused` (see, e.g.: https://html.spec.whatwg.org/multipage/media.html#event-media-playing). If you're keeping it for parity/clarity with the related tests, that's 👌